data_structures/linked_list.cpp

```diff
@@ -104,18 +126,43 @@ int main() {
         switch (choice) {
         case 1:
             std::cout << "\nEnter the element to be inserted : ";
-            std::cin >> x;
-            insert(x);
+            std::cin >> s;
+            
+            if(!isDigit(s))
+	    {
+		    std::cout<<"Wrong Input!\n";
+	    }
+            else
+            {
+            	x=toInt(s);
+            	insert(x);
+			}
             break;
         case 2:
             std::cout << "\nEnter the element to be removed : ";
-            std::cin >> x;
-            remove(x);
+            std::cin >> s;
+            if(!isDigit(s))
+	    {
+		    std::cout<<"Wrong Input!\n";
+	    }
+            else
+            {
+            	x=toInt(s);
+            	remove(x);
+			}
             break;
         case 3:
             std::cout << "\nEnter the element to be searched : ";
-            std::cin >> x;
-            search(x);
+            std::cin >> s;
+            if(!isDigit(s))
+	    {
+		    std::cout<<"Wrong Input!\n";
+	    }
+            else
+            {
+            	x=toInt(s);
+            	search(x);
+			}
             break;
         case 4:
             show();
```

Comment:
For other functions below, add `data_structures::` at the beginning of the function's name.
```suggestion
            data_structures::show();
```